mesonbuild/wrap/wrap.py

```diff
@@ -413,11 +414,11 @@ def get_git(self) -> None:
         if is_shallow and self.is_git_full_commit_id(revno):
             # git doesn't support directly cloning shallowly for commits,
             # so we follow https://stackoverflow.com/a/43136160
-            verbose_git(['init', self.directory], self.subdir_root, check=True)
+            verbose_git(['init', '-b', 'meson-dummy-branch', self.directory], self.subdir_root, check=True)
```

Comment:
Why do we need to create a branch here, can't it left detached? If that's only to avoid printing `Initialized empty Git repository in ...` then just use `quiet_git()` instead?

Reply:
Oh, did not know it would fallback the `master`, I thought it would fallback to detached mode. We actually checkout the commit later so we end up in the detached mode anyway.

Nitpicking: Maybe remove the dummy branch after the checkout?